packages/application/src/lab.ts

```diff
@@ -45,6 +42,24 @@ export class JupyterLab extends JupyterFrontEnd<ILabShell> {
     // Populate application info.
     this._info = { ...JupyterLab.defaultInfo, ...info };
 
+    this.restored = this.shell.restored
+      .then(() => {
+        // Backward compatibility
+        try {
+          // @ts-ignore
+          this.activateDeferredPlugins();
+        } catch (error) {
+          // no-op
+        }
+
+        if (this._info.deferred) {
+          this._info.deferred.matches.forEach(pluginID =>
+            this.activatePlugin(pluginID)
+          );
```

Comment:
Could we wrap this in a promise and display any error?

```suggestion
          Promise.all(this._info.deferred.matches.map(pluginID =>
            this.activatePlugin(pluginID)
          )).catch(error => { console.error('Error when activating customize list of deferred plugins:\n', error); });
```